### DISCUSSION TITLE
How can I delete input devices in cfclient?

### DISCUSSION THREAD

**Discussion by kaztateishi** [bitcraze/discussions#169] [Q&A] [answered]
I accidentally registered my keyboard as an input device and now the command based control part is always grayed out. How can I delete the input device so that I can test out the lighthouse system?
I tried to uninstall and re-install cfclient, but nothing's changed. I'm wondering there's a config data outside the directory. I'm using ubuntu 20.04 LTS.
*All things are working with my another computer (also Ubuntu 20.04), if I could just delete the input device config, then it's ideal.

Thanks for your help!

**Comment by krichardsson** [comment]
You can find all config files by using the `Settings/Open config folder` menu in the client. On ubuntu it should be /home/my-user/.config/cfclient
Not exactly sure which file you need to change, but worst case you can delete them all as they should be recreated with default values.

**Reply by kaztateishi** [reply]
Thanks for the reply @krichardsson !
Unfortunately, that did not work. Even when I delete all in the directory, cfclient still sees my keyboard and says "Using Normal mux with GIGABYTE USB-HID Keyboard Mouse... " (*My laptop's keyboard).
Any idea? Do I have to start over, maybe from OS install?

**Comment by knmcguire** [comment, marked as answer]
Hi! It sounds very similar to this issue somebody had on the previous forum:
https://forum.bitcraze.io/viewtopic.php?t=5083&start=10

So the short version is: 
In the terminal you will see something like this (but then the name of your keyboard device)

`INFO:cfclient.utils.input.mux:Adding device Microsoft X-Box 360 pad to MUX Normal
`

If you can close the cfclient, copy that name and add that to config.json like here:

`  "input_device_blacklist": "(VirtualBox|VMware|Microsoft X-Box 360 pad)",
`
and save it and open the cfclient again, your keyboard should be excluded.

>Make sure to fully close the cfclient before adding + saving this extra device to the config.json, because the cfclient will overwrite that file everytime it is closed.

**Reply by kaztateishi** [reply]
@knmcguire That worked! Thank you!
